### ISSUE TITLE
Behavior of `jax.experimental.array_api.var` w/ `correction > n`

### ISSUE THREAD

**Issue by mdhaber** [google/jax#21330]
### Description

`jax.experimental.array_api.var` does not follow the special case specified in the [array API documentation of `var`](https://data-apis.org/array-api/latest/API_specification/generated/array_api.var.html) for `correction > N`.

> Let `N` equal the number of elements over which to compute the variance.
>
> - If `N - correction` is less than or equal to 0, the variance is NaN.

```python3
import jax.experimental.array_api as xp
xp.var(xp.asarray([1.]), correction=1)  # Array(nan, dtype=float32)
xp.var(xp.asarray([]), correction=1)  # Array(-0., dtype=float32)
```

In most cases, `-0` is be the result of following the usual formula. Whether it is applicable for `correction > n` is debatable, but at least in the case above, the mean is undefined, so it might be desirable to update `jax.numpy`, too.

```python3
import jax.numpy as jnp
jnp.mean(jnp.asarray([]))  # Array(nan, dtype=float32)
jnp.var(jnp.asarray([]), ddof=1)  # Array(-0., dtype=float32)
```

For context, I ran into this when adding array API support to `scipy.stats.bartlett` (scipy/scipy#20751).

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.4.28
jaxlib: 0.4.28
numpy:  1.25.2
python: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
jax.devices (1 total, 1 local): [CpuDevice(id=0)]
process_count: 1
platform: uname_result(system='Linux', node='2e6238110dfb', release='6.1.85+', version='#1 SMP PREEMPT_DYNAMIC Sun Apr 28 14:29:16 UTC 2024', machine='x86_64')
```

**Comment by vfdev-5** [comment]
To return NaN in case whether `N - ddof <= 0`, we can simply return nan when the normalizer (which is essentially the size of input array) is negative:
```diff
  normalizer = lax.sub(normalizer, lax.convert_element_type(ddof, computation_dtype))
  result = sum(centered, axis, dtype=computation_dtype, keepdims=keepdims, where=where)
  return _where(normalizer > 0, lax.div(result, normalizer).astype(dtype), np.nan)
```
